PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift

```diff
@@ -64,14 +69,27 @@ class SearchViewModel: ObservableObject {
         let term = searchTerm.trimmingCharacters(in: .whitespaces).lowercased()
         let shouldShowUpsell = !isPremium && selectedScope == .all
         guard !shouldShowUpsell else { return }
-
-        emptyState = searchResultState()
+        submitSearch(with: term)
         showRecentSearches = false
 
         guard isPremium, !term.isEmpty else { return }
         recentSearches = updateRecentSearches(with: term)
     }
 
+    func clear() {
+        searchResults = []
+        subscriptions = []
+    }
+
+    private func submitSearch(with term: String) {
+        clear()
+        source.search(for: term, scope: selectedScope)
+        source.events.receive(on: DispatchQueue.main).sink { [weak self] event in
```

Comment:
oh this code has been outdated, but yes you're right! the updated changes should have it subscribed first. This is the new code using `searchService` instead of `source`

```
searchService.results.receive(on: DispatchQueue.main).sink { [weak self] items in
             guard let self else { return }
             self.searchResults = items.compactMap { SearchItem(item: $0) }
         }.store(in: &subscriptions)
         Task {
             await searchService.search(for: term, scope: selectedScope)
         }
```